ThirdPartyAdapters/chartboost/chartboost/src/main/java/com/google/ads/mediation/chartboost/ChartboostSingleton.java

```diff
@@ -177,26 +207,53 @@ public static void startChartboostRewardedVideo(
                     "An ad has already been requested for the location: " + location);
             adapterDelegate.didFailToLoadRewardedVideo(location,
                     CBError.CBImpressionError.NO_AD_FOUND);
+            mRewardedDelegates.remove(location);
             return;
         }
 
         // Add this adapter delegate to mAdapterDelegates so that the events from
         // Chartboost SDK can be forwarded.
         addRewardedDelegate(location, adapterDelegate);
-        startChartboost((Activity) context, adapterDelegate.getChartboostParams(), adapterDelegate);
+        startChartboost(context, adapterDelegate.getChartboostParams(), adapterDelegate, null);
+    }
+
+    /**
+     *  This method will initialize Chartboost SDK for banner ads and return whether or not
+     *  it successfully initialized.
+     * @param context
+     * @param adapterDelegate
+     * @param bannerListener
+     */
+    public static void startChartboostBanner(Context context,
+                                             AbstractChartboostAdapterDelegate adapterDelegate,
+                                             ChartboostBannerListener bannerListener) {
+        String location = adapterDelegate.getChartboostParams().getLocation();
+
+        // Checks if an ad has already been sent for caching for the requested location, and fail
+        // the ad request if it is.
+        WeakReference<ChartboostBanner> delegate = getChartboostBanner(location);
+        if (delegate != null && delegate.get() != null) {
+            Log.w(ChartboostMediationAdapter.TAG, "An banner ad has already been requested for the location: " + location);
+            adapterDelegate.didFailToLoadInterstitial(location, CBError.CBImpressionError.NO_AD_FOUND);
+            mBannerMap.remove(location);
+            return;
+        }
+
+        startChartboost(context, adapterDelegate.getChartboostParams(), adapterDelegate, bannerListener);
     }
 
     /**
      * This method will initialize the Chartboost SDK if it is not already initialized and set its
      * delegate.
      *
-     * @param activity        required to initialize {@link Chartboost}.
+     * @param context        required to initialize {@link Chartboost}.
      * @param params          The Chartboost params containing server parameters and network extras
      *                        to be used to start {@link Chartboost}.
      * @param adapterDelegate The adapter delegate to which to forward initialization callbacks.
      */
-    private static void startChartboost(Activity activity, ChartboostParams params,
-                                        AbstractChartboostAdapterDelegate adapterDelegate) {
+    private static void startChartboost(Context context, ChartboostParams params,
+                                        AbstractChartboostAdapterDelegate adapterDelegate,
+                                        ChartboostBannerListener bannerListener) {
         if (mIsChartboostInitializing) {
```

Comment:
Seems like we are silently failing if trying to load a banner while Chartboost is initializing. I think we should allow this operation. The first banner (the one that triggers the initialization when the SDK wasn't even started to initializing) is allowed after all.

Reply:
I think this needs to stay here as it used to be cause every single ad is coming through this function and we don't wanna initialise if it is already in progress.

Reply:
I understand that you don't want to initialize, but you are doing more than that here. You are preventing the request from going ahead (even though you allow the first request before the sdk was asked to initialize!), and moreover, the request is failing silently. This will cause timeouts with AdMob which leads to other problems.

Reply:
I was misunderstanding the issue. Solved